lib/plugin.js

```diff
@@ -42,24 +42,26 @@ ManifestPlugin.prototype.apply = function(compiler) {
     var stats = compilation.getStats().toJson();
     var manifest = {};
 
-    _.merge(manifest, compilation.chunks.reduce(function(memo, chunk) {
-      var chunkName = chunk.name ? chunk.name.replace(this.opts.stripSrc, '') : null;
-
-      // Map original chunk name to output files.
-      // For nameless chunks, just map the files directly.
-      return chunk.files.reduce(function(memo, file) {
-        // Don't add hot updates to manifest
-        if (file.indexOf('hot-update') >= 0) {
+    _.merge(manifest, compilation.chunks
+      .filter(chunk => chunk.isInitial ? chunk.isInitial() : chunk.initial)
+      .reduce(function(memo, chunk) {
```

Comment:
It looks like if we use `(memo, chunk) => {...}` we get the right context and we can skip `.bind(this)`